markhor_tracks/src/markhor_hw_interface.cpp

```diff
@@ -52,36 +52,97 @@ void MarkhorHWInterface::setupCTREDrive()
+    rear_left_drive->ConfigSelectedFeedbackSensor(FeedbackDevice::CTRE_MagEncoder_Relative , 0, timeout_ms_);
+    rear_left_drive->SetSensorPhase(true);
+    rear_left_drive->ConfigSupplyCurrentLimit(current_limit_config);
+    rear_left_drive->ConfigNominalOutputForward(0, timeout_ms_);
+    rear_left_drive->ConfigNominalOutputReverse(0, timeout_ms_);
+    rear_left_drive->ConfigAllowableClosedloopError(0, 0, timeout_ms_);
+    rear_left_drive->SelectProfileSlot(0, 0);
+    rear_left_drive->Config_kF(0, 0, timeout_ms_);
+    rear_left_drive->Config_kP(0, 0.3, timeout_ms_);
+    rear_left_drive->Config_kI(0, 0.005, timeout_ms_);
+    rear_left_drive->Config_kD(0, 0, timeout_ms_);
   }
   if (nh.getParam("/markhor/markhor_tracks_node/front_right", drive_fr_id) == true)
   {
     front_right_drive = std::make_unique<TalonSRX>(drive_fr_id);
-    front_right_drive->SetNeutralMode(NeutralMode::Brake);
+    front_right_drive->SetNeutralMode(NeutralMode::Coast);
     front_right_drive->SetInverted(true);
+    front_right_drive->ConfigFactoryDefault();
+    front_right_drive->ConfigSelectedFeedbackSensor(FeedbackDevice::CTRE_MagEncoder_Relative , 0, timeout_ms_);
+    front_right_drive->SetSensorPhase(true);
+    front_right_drive->ConfigSupplyCurrentLimit(current_limit_config);
+    front_right_drive->ConfigNominalOutputForward(0, timeout_ms_);
+    front_right_drive->ConfigNominalOutputReverse(0, timeout_ms_);
+    front_right_drive->ConfigAllowableClosedloopError(0, 0, timeout_ms_);
+    front_right_drive->SelectProfileSlot(0, 0);
+    front_right_drive->Config_kF(0, 0, timeout_ms_);
+    front_right_drive->Config_kP(0, 0.3, timeout_ms_);
+    front_right_drive->Config_kI(0, 0.005, timeout_ms_);
+    front_right_drive->Config_kD(0, 0, timeout_ms_);
   }
   if (nh.getParam("/markhor/markhor_tracks_node/rear_right", drive_rr_id) == true)
   {
     rear_right_drive = std::make_unique<TalonSRX>(drive_rr_id);
-    rear_right_drive->SetNeutralMode(NeutralMode::Brake);
+    rear_right_drive->SetNeutralMode(NeutralMode::Coast);
+    rear_right_drive->ConfigFactoryDefault();
+    rear_right_drive->ConfigSelectedFeedbackSensor(FeedbackDevice::CTRE_MagEncoder_Relative , 0, timeout_ms_);
+    rear_right_drive->SetSensorPhase(false);
+    rear_right_drive->ConfigSupplyCurrentLimit(current_limit_config);
+    rear_right_drive->ConfigNominalOutputForward(0, timeout_ms_);
+    rear_right_drive->ConfigNominalOutputReverse(0, timeout_ms_);
+    rear_right_drive->ConfigAllowableClosedloopError(0, 0, timeout_ms_);
+
+    rear_right_drive->SelectProfileSlot(0, 0);
+    rear_right_drive->Config_kF(0, 0, timeout_ms_);
+    rear_right_drive->Config_kP(0, 0.1, timeout_ms_);//TODO : Set this value to 0.3 when encoder is fixed
+    rear_right_drive->Config_kI(0, 0.002, timeout_ms_);//TODO : Set this value to 0.005 when encoder is fixed
+    rear_right_drive->Config_kD(0, 0, timeout_ms_);
+
+    front_left_drive->Set(ControlMode::Velocity, 0);
```

Comment:
Dit moi si je me trompe, mais si tu veux faire en sorte que lorsque tu initialises les drives tu veux que leur vitesse soit à zéro, il faut que tu ajoutes la ligne `ctre::phoenix::unmanaged::FeedEnable(100);` pour que les encodeurs envoient des commandes pour les `100` prochaines millisecondes. https://www.ctr-electronics.com/downloads/api/java/html/classcom_1_1ctre_1_1phoenix_1_1unmanaged_1_1_unmanaged.html#a711cfdfef527ff4324af0b442de6efed

Reply:
Aussi, je préfèrerai qu'on ne mette pas ces commandes dans le dernier `if` statement. En fait, ça serait plus correct de checker quel drive existe ou ont été initialisés et mettre à zéro ces drives là.

Reply:
Pour le feed enable, no idea what it does, faut que je check ça, pour le if faut que j'analyse ça un peu plus, t'a probablement raison

Reply:
ok c'est vraiment dumb ce que j'ai fait, on garde le set à 0 mais on le mets chacun dans son if respectif?

Reply:
> on garde le set à 0 mais on le mets chacun dans son if respectif?

Ouais, c'est probablement mieux que ce que j'avais en tête, mais va falloir que tu mets un feed enable dans chaque if aussi. Le feedEnable est la fonction qui dit aux drives d'accepter pendant X millisecondes les commandes qu'on envoie. Va voir la fonction `write` c'est ça qu'on fait.

Reply:
normalement c'est fixed avec mes 2 derniers commit, reste a tester, j'ai use timeout_ms comme pour les autres fonctions pour standardiser, à moins que j'ai mal compris la fonction feedenable

Reply:
Ehhh ouin tu peux l'utilisé, mais j'aurai tendance à dire que c'est un peu "weird", pcq le timeout_ms c'est plus utiliser pour si tu n'as pas de réponse de la drive ne fait pas l'opération. Tant dit que dans le feedEnable c'est plus pendant X nombre de temps, tu peux appliquer les commandes envoyées. Donc, conceptuellement, ce n'est pas vraiment la même signification. J'aurai tendance à simplement ajouter une variable de plus qui est plus explicative, genre `feedenable_timeframe` ou quelque chose du genre.

Reply:
ouais ce serrait mieux, faut switch à ça